File: Calculator.cpp
// Example program
#include <iostream>
#include <string>
#include <vector>
#include <chrono>
#include <ctime>
#include <fstream>

using namespace std;

// Calculator  class
class Calculator {
    // History of results
private: vector<float> history;

    // Constructor for loading results from file
    public: Calculator(string fileName) {
        loadHistoryFromFile(fileName);
        print("Calculator was created");
    }
    
    Calculator() {
        print("Calculator was created");
    }

    // Prints text with current date
    public: void print(string text) {
        std::time_t date = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
        std::cout << "# " <<  std::ctime(&date) << " # " << text;
    }

    // Adds two numbers
    public: float addition(float number1, float number2) {
        float result = number1 + number2;
        print("Result od addition is " + to_string(result));
        addToHistory(result);
        return result;
    }

    // Subtracts two numbers
    public: float substraction(float number1, float number2) {
        float result = number1 - number2;
        print("Result od substraction is " + to_string(result));
        addToHistory(result);
        return result;
    }


    // Multiplies two numbers
    public: float multiplication(float number1, float number2) {
        float result = number1 * number2;
        print("Result od multiplication is " + to_string(result));
        addToHistory(result);
        return result;
    }

    // Divides two numbers
    public: float division(float number1, float number2) {
        float result = number1 / number2;
        print("Result od division is " + to_string(result));
        addToHistory(result);
        return result;
    }

    // Adds result to history array
    private: void addToHistory(float item) {
        history.push_back(item);
    }

    // Print the whole history
    public: void printHistory() {
        for (float result: history) {
            print(to_string(result));
        }
    }

    // Compare two numbers
    public: void compareNumbers(float number1, float number2) {
        if (number1 > number2) {
            print(to_string(number1) + " is bigger than " + to_string(number2));
        } else if (number2 > number1) {
            print(to_string(number1) + " is smaller than " + to_string(number2));
        } else {
            print(to_string(number1) + " is the same as " + to_string(number2));
        }
    }

    // Adds results to history
    public: void loadHistoryFromFile(string fileName) {
        try {
            std::ifstream file(fileName);
            string readLine = "";
            while (getline(file, readLine)) {
                addToHistory(stof(readLine));
            }
            file.close();
            print("History form file was loaded!");
        } catch (const std::exception& e) {
            print("File could not be loaded! Because:" + std::string(e.what()));
        }
    }

    // Returns if number is odd
    public: bool isOdd(float number) {
        bool boolIsOdd = (number/2) != 0;
        switch (boolIsOdd) {
            case true:
                print(to_string(number) + " is odd!");
                return true;
            case false:
                print(to_string(number) + " is not odd!");
                return false;
        }
    }
};

File: php/Calculator.php
<?php
class Calculator {
	// History of $results
	private $history = array();
	
	// Constructor for loading results from file
	public function __construct($fileName = '') {
		$this->loadHistoryFromFile($fileName);
		$this->print("Calculator was created");
	}
	
	// Prints text with current date
	public function print($text) {
		$date = date('m/d/Y h:i:s a', time());
	    echo "# " . $date . " # "  . $text;
	}
	
	// Adds two numbers
	public function addition($number1, $number2) {
		$result = $number1 + $number2;
		$this->print("Result od addition is " . $result);
		$this->addToHistory($result);
		return $result;
	}
	
	// Subtracts two numbers
	public function substraction( $number1, $number2) {
		$result = $number1 - $number2;
		$this->print("Result od substraction is " . $result);
		$this->addToHistory($result);
		return $result;
	}
	
	
	// Multiplies two numbers
	public function multiplication( $number1, $number2) {
		$result = $number1 * $number2;
		$this->print("Result od multiplication is " . $result);
		$this->addToHistory($result);
		return $result;
	}
	
	// Divides two numbers
	public function division( $number1, $number2) {
		$result = $number1 / $number2;
		$this->print("Result od division is " . $result);
		$this->addToHistory($result);
		return $result;
	}
	
	// Adds $result to history array
	private function addToHistory($item) {
        array_push($this->history, $item);
	}
	
	// Print the whole history
	public function printHistory() {
		foreach ( $this->history as $result) {
			$this->print($result);
		}
	}
	
	// Compare two numbers
	public function compareNumbers( $number1, $number2) {
		if ($number1 > $number2) {
			$this->print($number1 . " is bigger than " . $number2);
		} else if ($number2 > $number1) {
			$this->print($number1 . " is smaller than " . $number2);
		} else {
			$this->print($number1 . " is the same as " . $number2);
		}
	}
	
	// Adds $results to history
	public function loadHistoryFromFile($fileName) {
        if($fileName !== '')
        try {
			$file = fopen($fileName, "r");
            $readLine = "";
            while (($readLine = fgets($file)) !== false) {
                $this->addToHistory($readLine);
            }
            fclose($file);
            $this->print("History form file was loaded!");
		} catch (Exception $e) {
			$this->print("File could not be loaded! Because:" . $e);
		}
	}
	
	// Returns if number is odd
	public function isOdd( $number) {
		$boolIsOdd = ($number/2) != 0;
		switch ($boolIsOdd) {
			case true:
				$this->print($number . " is even!");
				return true;
			case false: 
				$this->print($number . " is not odd!");
				return false;
		}
	}
}